### ISSUE TITLE
[Workflow] GRPC connection to workflow runtime doesn't self-heal when app restarts

### ISSUE THREAD

**Issue by olitomlinson** [dapr/dotnet-sdk#1293]
cc @philliphoff 

runtime 1.13.2 (not tried any other versions)

## Expected Behavior

The grpc connection to the workflow runtime will reestablish after the app process (not dapr process) crashes and is restarted.

## Actual Behavior

The grpc connection to the workflow runtime **does not** reestablish after the app process (not dapr process) crashes and is restarted.

## Steps to Reproduce the Problem

Pull down my repro here https://github.com/olitomlinson/dapr-workflow-examples

1. run `docker compose build`
2. run `docker compose up`
3. stop the `app` container in compose - it will be named something like `workflow-app-a-1`
4. start the `app` container in compose
5. observe the logs in `workflow-app-a-1` and you will see the following error repeating forever : 

`The gRPC server for Durable Task gRPC worker is unavailable. Will continue retrying.`

## Release Note

RELEASE NOTE:

**Comment by cgillum** [comment]
This may have been fixed already in 1.14 as part of pulling in some fixes in `durabletask-go`. @olitomlinson are you able to verify?

**Comment by olitomlinson** [comment]
> This may have been fixed already in 1.14 as part of pulling in some fixes in `durabletask-go`. @olitomlinson are you able to verify?

Still an issue in 1.14.4

**Comment by famarting** [comment]
I find this confusing. For the go-sdk I made the client to infinitely retry the worker connection to dapr, and I think we should have that behavior on every SDK, I believe python already has it.

**Comment by WhitWaldo** [comment]
@olitomlinson Do you know if this is still an issue with the 1.15 RC?

**Comment by olitomlinson** [comment]
> [@olitomlinson](https://github.com/olitomlinson) Do you know if this is still an issue with the 1.15 RC?

@WhitWaldo yes, just tested on `1.15.0-rc.7`, and it still exhibits the same behavior :(

**Comment by olitomlinson** [comment]
Still not self-healing in runtime 1.16.0-rc.25/ dotnet sdk 1.16.0-rc15

**Comment by JoshVanL** [comment]
> @JoshVanL mac

Did you upgrade the OS by chance?

I think this is an OS DNS breaking change we need to handle.
